File: kiki616/ABC/B/095.py
N,X = map(int,input().split())
M = []
cnt = 0
for i in range(N):
    m = int(input())
    M.append(m)
    X -= m
    cnt += 1

while True:
    if X >= min(M):
        X -= min(M)
        cnt += 1
    else:
        break
print(cnt)

File: OCH_Tao/ABC/B/095.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
let [N, X] = ((input.shift()).split(" ")).map(x => parseInt(x, 10));
const M = input.map(x => parseInt(x, 10));
M.sort((a, b) => a - b);
let cnt = N;
X -= M.reduce((a, b) => a + b);
while (X >= M[0]) {
  X -= M[0];
  cnt++;
}
console.log(cnt);